bridges/otelzap/zap_encoder.go

```diff
@@ -0,0 +1,233 @@
+		log.KeyValue{
+			Key:   k,
+			Value: assignUintValue(v),
+		})
+}
+
+// It calls this method if interface cannot be mapped to supported zap types
+// converts everything to a JSON string.
+func (m *ObjectEncoder) AddReflected(k string, v interface{}) error {
+	enc := json.NewEncoder(m)
+	if err := enc.Encode(v); err != nil {
+		return err
+	}
+	m.AddString(k, m.reflectval.AsString())
+	return nil
+}
+
+// Implements Write method to which json encoder writes to.
+// Used by AddReflected method.
+func (m *ObjectEncoder) Write(p []byte) (n int, err error) {
+	m.reflectval = log.StringValue(string(p))
+	return
+}
+
+// TODO:
+// OpenNamespace opens an isolated namespace where all subsequent fields will
+// be added.
+func (m *ObjectEncoder) OpenNamespace(k string) {
+	// ns := make([]log.KeyValue, 0)
+	// // m.cur expects both key and value
+	// // "Namespace" as value here should be confirmed
+	// m.cur = append(m.cur, log.String(k, "Namesspace"))
+	// m.cur = ns
+}
+
+func (m *ObjectEncoder) AddComplex64(k string, v complex64) { m.AddComplex128(k, complex128(v)) }
+func (m *ObjectEncoder) AddInt32(k string, v int32)         { m.AddInt64(k, int64(v)) }
+func (m *ObjectEncoder) AddInt16(k string, v int16)         { m.AddInt64(k, int64(v)) }
+func (m *ObjectEncoder) AddInt8(k string, v int8)           { m.AddInt64(k, int64(v)) }
+func (m *ObjectEncoder) AddTime(k string, v time.Time)      { m.AddInt64(k, v.UnixNano()) }
+func (m *ObjectEncoder) AddUint(k string, v uint)           { m.AddUint64(k, uint64(v)) }
+func (m *ObjectEncoder) AddUint32(k string, v uint32)       { m.AddInt64(k, int64(v)) }
+func (m *ObjectEncoder) AddUint16(k string, v uint16)       { m.AddInt64(k, int64(v)) }
+func (m *ObjectEncoder) AddUint8(k string, v uint8)         { m.AddInt64(k, int64(v)) }
+func (m *ObjectEncoder) AddUintptr(k string, v uintptr)     { m.AddUint64(k, uint64(v)) }
+
+// ArrayEncoder implements zapcore.ArrayEncoder.
+type ArrayEncoder struct {
+	elems []log.Value
+}
+
+func (a *ArrayEncoder) AppendArray(v zapcore.ArrayMarshaler) error {
+	enc := &ArrayEncoder{}
+	err := v.MarshalLogArray(enc)
+	a.elems = append(a.elems, enc.elems...)
+	return err
+}
+
+func (a *ArrayEncoder) AppendObject(v zapcore.ObjectMarshaler) error {
+	// passing 0 here - we do not know object's length
+	// a minimum buffer capacity can be agreed upon?
+	m := NewObjectEncoder(0)
+	err := v.MarshalLogObject(m)
+	a.elems = append(a.elems, log.MapValue(m.cur...))
+	return err
+}
+
+func (a *ArrayEncoder) AppendReflected(v interface{}) error {
+	enc := json.NewEncoder(a)
+	return enc.Encode(v)
+}
+
+func (a *ArrayEncoder) AppendComplex128(v complex128) {
+	stringValue := `"` + strconv.FormatComplex(v, 'f', -1, 64) + `"`
+	a.elems = append(a.elems, log.StringValue(stringValue))
+}
+
+func (a *ArrayEncoder) AppendFloat32(v float32) {
+	// preserves float32 value
+	value, _ := strconv.ParseFloat(strconv.FormatFloat(float64(v), 'f', -1, 32), 64)
+	a.AppendFloat64(value)
```

Comment:
Why not

```suggestion
	a.AppendFloat64(float64(v))
```

Reply:
Because float64(v) doesn't preserve the float32 value as is. This is the corresponding test that will fail 
https://github.com/open-telemetry/opentelemetry-go-contrib/blob/9579fa0c261ceeadae8e2cd499df1034c5b1f205/bridges/otelzap/zap_test.go#L103

Fo ex: 
```
a := float32(3.14)
fmt.Println(float64(a))
```
will return `3.140000104904175`

Reply:
Use https://pkg.go.dev/github.com/stretchr/testify/assert#InDelta for asserting the result.